mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2671,10 +2671,15 @@ where
             if let Some(gas) = gas_fee {
                 tx_builder = tx_builder.with_gas_fee(gas);
             }
-            let (tx, data) = tx_builder
-                .build()
-                .await
-                .mm_err(|e| TradePreimageError::from_generate_tx_error(e, ticker, decimals, is_amount_upper_bound))?;
+            let (tx, data) = tx_builder.build().await.mm_err(|e| {
+                TradePreimageError::from_generate_tx_error(
+                    e,
+                    ticker,
+                    coin.platform_ticker().to_string(),
```

Comment:
As I said in the first comment, `TradePreimageError::from_generate_tx_error` doesn't need to take `platform_coin`.
But we **need** to pass a correct ticker here. So this is what I consider as a solution:
I'd add the `ticker` additional argument to the `utxo_common:: preimage_trade_fee_required_to_send_outputs` function. The `ticker` argument should be used instead of
https://github.com/KomodoPlatform/atomicDEX-API/blob/4366141c8a7a5d6d3abac87df8827f7cae53d11c/mm2src/coins/utxo/utxo_common.rs#L2648

So you need to find all places where we need to pass `coin.platform_ticker()` instead of `coin.ticker()` to the `utxo_common:: preimage_trade_fee_required_to_send_outputs` function. And this will be a solution.
If you have other ideas, please feel free to share them to discuss 🙂

Reply:
yes cool, this is also a very good solution from my side ...thoughts @shamardy

Reply:
I agree with @sergeyboyko0791